conans/client/generators/cmake_find_package_multi.py

```diff
@@ -260,8 +260,8 @@ class CMakeFindPackageMultiGenerator(CMakeFindPackageGenerator):
 
     def __init__(self, conanfile):
         super(CMakeFindPackageMultiGenerator, self).__init__(conanfile)
-        self.configurations = ["Release", "RelWithDebInfo", "MinSizeRel", "Debug"]
         self.configuration = str(self.conanfile.settings.build_type)
+        self.configurations = [v for v in conanfile.settings.build_type.values_range if v != "None"]
```

Comment:
Maybe we can turn this into a property and prevent people from assigning/appending to it... or make it a private attribute.


```python
@property
def configurations(self):
    return [v for v in conanfile.settings.build_type.values_range if v != "None"]
```

Reply:
It is still possible to add new configurations, that are not defined in ``settings.yml``, for example "ReleaseShared", that is defined from settings build_type + option shared. There is a test for it, and the only way is that recipes are able to explicitly add it. 

The moment the generator is fully dynamic, it can be removed, but atm, it will make that test fail.